cli/args/config_file.rs

```diff
@@ -357,13 +362,30 @@ impl SerializedLintConfig {
   }
 }
 
+pub trait ContainsFilesConfig {
+  fn get_files_config(&self) -> FilesConfig;
+}
+
 #[derive(Clone, Debug, Default)]
 pub struct LintConfig {
   pub rules: LintRulesConfig,
   pub files: FilesConfig,
   pub report: Option<String>,
 }
 
+impl ContainsFilesConfig for LintConfig {
+  fn get_files_config(&self) -> FilesConfig {
+    self.files.clone()
+  }
+}
+
+#[derive(Clone, Debug)]
+pub struct FinalLintConfig {
```

Comment:
`SerializedLintConfig`, `LintConfig`, and `FinalLintConfig` seems like too many types. Is it possible to just have `SerializedLintConfig` -> `LintConfig` and resolve everything in `to_lint_config`? (Same goes for the other types)

Reply:
Do you mind if I take over this PR to get it merged? Reviewing it would take too much back and forth.

Reply:
Sure, feel free to take it.